### PR TITLE
Feature/big table improvements

### LINKED ISSUE #130: Order herd select fields by number
Select fields that display herds should be ordered by number and not alphabetically.

### LINKED ISSUE #132: Allow a subset of columns by default in FilterTable
Currently the FilterTable crashes if you add columns that weren't in there when the Table is created.

### PR DESCRIPTION
**Pull request by norling**
This still doesn't allow column sorting, but it handles the rest of the table a bit better, and doesn't need the material-table dependency anymore. I'm gonna keep looking at the column sorting, but this started to grow to large, so I figured that it was better to make a PR with just the new table!

Closes #130 
Closes #132

### CODE REVIEW

**Review by jhagberg — COMMENTED** [review]
Get a snowpack error on the dev server
```
[snowpack] installing dependencies...
herdbook-frontend-devel_1  | [snowpack] Package "@date-io/date-fns" not found. Have you installed it? 

```

**Review by jhagberg — CHANGES_REQUESTED** [review]
Things I notice when testing Looks nicer now I think. Search is a bit better looks like it renders faster also.

problems or questions:

- Cant hide Dead animals on Herd page. 
- The death date is set to "11/5/2020" for all animals both in hed view and in genebank view.
- Opening a herd on new page gives nginx error 404

File: frontend/src/filter_table.tsx [hunk 1]
```diff
@@ -72,74 +40,112 @@ const useStyles = makeStyles({
     color: 'blue',
     textDecoration: 'underline',
     cursor: 'pointer',
+  },
+  hidden: {
```
Comment (danr) [inline frontend/src/filter_table.tsx]: What's happening here? Why not `display: none` or `visibility:hidden`?

File: frontend/src/filter_table.tsx [hunk 2]
```diff
@@ -165,25 +174,150 @@ export function FilterTable({individuals, title = '', filters = [],
...
+   * subfield to use for sorting. `sortAs` can be `number`, `numbers` or `date`
+   * to force a column to be sorted as a particular type.
+   * - `date` fields will be cast as dates using `new Date(<value)`
+   * - `number` fields will be cast as numbers by removing all non-numerical
+   *   characters. ex. `G111-222` -> `111222`
+   * - `numbers` fields are cast as numbers, but using the first set of numbers
+   *   as the integer part, and the remaning as decimals.
+   *   ex. `G111-222-333` -> `111.222333`.
+   */
+  const sortedIndividuals = React.useMemo(() => {
```
Comment (danr) [inline frontend/src/filter_table.tsx:302]: I would go for putting the sort function on the top level or in a separate module so it's easier to test.

File: frontend/src/filter_table.tsx [hunk 3]
```diff
@@ -72,74 +40,112 @@ const useStyles = makeStyles({
...
+  search: {
+    float: 'right',
   }
 });
 
+interface Column {
+  field: keyof Individual;
+  label: string;
+  sortBy?: string;
+  sortAs?: string;
```
Comment (danr) [inline frontend/src/filter_table.tsx]: A bit unsure what the difference between sortBy and sortAs are here

File: frontend/src/filter_table.tsx [hunk 4]
```diff
@@ -155,7 +161,10 @@ export function FilterTable({individuals, title = '', filters = [],
     setColumns(newColumns)
   }
 
-  const filteredIndividuals = () => {
+  /**
+   * Memoized list holding all individuals that pass the current filter(s).
+   */
+  const filteredIndividuals: Individual[] = React.useMemo(() => {
```
Comment (danr) [inline frontend/src/filter_table.tsx:280]: I would put this in a top-level function as well to make testing easier.

**Review by danr — APPROVED** [review]
Some comments on the shape of the code

File: frontend/src/filter_table.tsx (hunk 3 above)
Comment (norling) [reply frontend/src/filter_table.tsx]: ah - should have commented on that. `sortBy` is to sort the column by a sub-field when the value is an object, `sortAs` is to coerce the value to a new type before sorting!

File: frontend/src/filter_table.tsx (hunk 1 above)
Comment (norling) [reply frontend/src/filter_table.tsx]: ah! forgot to change the name ^^ That class is for the sorting arrow.

**Review by danr — APPROVED** [review]
Alright! :sunglasses:

**Review by jhagberg — APPROVED** [review]
LGTM

**Pull request merged**